tools/fetch-spectacle.js

```diff
@@ -0,0 +1,435 @@
+  );
+
+  code = code.replace(
+    '{class:"bi-exclamation-triangle", style:{"color":"red", "margin-left":"5px"}}, " Parse error.")',
+    '{style:{"color":"red", "margin-left":"5px", "font-weight":"bold"}}, "⚠ Parse error.")'
+  );
+
+  code = code.replace('class: "bi bi-link fancy",', '');
+  code = code.replace('class: "bi bi-gear",', '');
+  code = code.replace('class: "bi bi-trash",', '');
+
+  code = code.replace(
+    'function loadSpecBox(hidden){\n    let loadFailedErrMsg = "Error fetching spec from given URL. Make sure the link is to a raw TLA+ file.";',
+    'function loadSpecBox(hidden){\n    if (window.SPECTACLE_VSCODE_MODE) {\n        return null;\n    }\n    let loadFailedErrMsg = "Error fetching spec from given URL. Make sure the link is to a raw TLA+ file.";'
+  );
+
+  code = code.replace(
+    '    loadSpecFromPath(model.specPath);\n}\n',
+    '    if (!window.SPECTACLE_VSCODE_MODE) {\n        loadSpecFromPath(model.specPath);\n    }\n}\n'
+  );
+
+  if (!code.includes('spectacle:webview-ready')) {
+    code = code.replace(
+      "    document.addEventListener('keydown', handleKeyboardNavigation);",
+      "    document.addEventListener('keydown', handleKeyboardNavigation);\n\n    if (vscodeApi) {\n        vscodeApi.postMessage({ type: 'spectacle:webview-ready' });\n    }"
+    );
+  }
+
+  const routerOriginal = '    // m.mount(root, App)\n    m.route(root, "/home",\n        {\n            "/home": App,\n            "/eval_debug_graph": EvalDebugGraph,\n        });';
+  const routerPatched = `    const disableSpectacleRouter = Boolean(typeof window !== "undefined" && window.SPECTACLE_ROUTER_DISABLE_HISTORY);
+    if (disableSpectacleRouter) {
+        let __spectacleRouteShimState = {};
+        if (typeof m.route !== "function") {
+            m.route = function () {};
+        }
+        if (typeof m.redraw !== "function") {
+            m.redraw = function () {};
+        }
+        m.route.param = function (key) {
+            if (typeof key === "string") {
+                return __spectacleRouteShimState[key];
+            }
+            return __spectacleRouteShimState;
+        };
+        m.route.set = function (_path, params) {
+            if (params && typeof params === "object") {
+                __spectacleRouteShimState = params;
+            } else {
+                __spectacleRouteShimState = {};
+            }
+            if (typeof m.redraw === "function") {
+                m.redraw();
+            }
+        };
+        m.route.get = function () {
+            return "/home";
+        };
+        m.mount(root, App);
+    } else {
+        m.route(root, "/home",
+            {
+                "/home": App,
+                "/eval_debug_graph": EvalDebugGraph,
+            });
+    }`;
+  if (code.includes(routerOriginal)) {
+    code = code.replace(routerOriginal, routerPatched);
+  }
+
+  fs.writeFileSync(appPath, code);
+}
+
+function patchEvalJs(destDir) {
```

Comment:
Patching Spectacle's evaluator to support TLC!Assert should instead be done in dedicated upstream PR.